<Type Name="StackOverflowException" FullName="System.StackOverflowException">
  <Metadata>
    <Meta Name="ms.openlocfilehash" Value="bb807895a73fd32279843a0dc0c00f2b0ed44976" />
    <Meta Name="ms.sourcegitcommit" Value="d877ae76e9e11799bf919379507239e2c4072742" />
    <Meta Name="ms.translationtype" Value="HT" />
    <Meta Name="ms.contentlocale" Value="de-DE" />
    <Meta Name="ms.lasthandoff" Value="08/09/2018" />
    <Meta Name="ms.locfileid" Value="39809691" />
  </Metadata>
  <TypeSignature Language="C#" Value="public sealed class StackOverflowException : SystemException" />
  <TypeSignature Language="ILAsm" Value=".class public auto ansi serializable sealed beforefieldinit StackOverflowException extends System.SystemException" />
  <TypeSignature Language="DocId" Value="T:System.StackOverflowException" />
  <TypeSignature Language="VB.NET" Value="Public NotInheritable Class StackOverflowException&#xA;Inherits SystemException" />
  <TypeSignature Language="C++ CLI" Value="public ref class StackOverflowException sealed : SystemException" />
  <TypeSignature Language="F#" Value="type StackOverflowException = class&#xA;    inherit SystemException" />
  <AssemblyInfo>
    <AssemblyName>mscorlib</AssemblyName>
    <AssemblyVersion>1.0.5000.0</AssemblyVersion>
    <AssemblyVersion>2.0.0.0</AssemblyVersion>
    <AssemblyVersion>2.0.5.0</AssemblyVersion>
    <AssemblyVersion>4.0.0.0</AssemblyVersion>
  </AssemblyInfo>
  <AssemblyInfo>
    <AssemblyName>netstandard</AssemblyName>
    <AssemblyVersion>2.0.0.0</AssemblyVersion>
  </AssemblyInfo>
  <AssemblyInfo>
    <AssemblyName>System.Runtime</AssemblyName>
    <AssemblyVersion>4.2.0.0</AssemblyVersion>
    <AssemblyVersion>4.2.1.0</AssemblyVersion>
  </AssemblyInfo>
  <Base>
    <BaseTypeName>System.SystemException</BaseTypeName>
  </Base>
  <Interfaces />
  <Attributes>
    <Attribute FrameworkAlternate="netframework-2.0;netframework-3.0;netframework-3.5;netframework-4.0;netframework-4.5;netframework-4.5.1;netframework-4.5.2;netframework-4.6;netframework-4.6.1;netframework-4.6.2;netframework-4.7;netframework-4.7.1;netframework-4.7.2;xamarinandroid-7.1;xamarinios-10.8;xamarinmac-3.0">
      <AttributeName>System.Runtime.InteropServices.ComVisible(true)</AttributeName>
    </Attribute>
  </Attributes>
  <Docs>
    <summary>
      <span data-ttu-id="d6af0-101">Die Ausnahme, die bei einem Überlauf des Ausführungsstapels durch zu viele geschachtelte Methodenaufrufe ausgelöst wird.</span>
      <span class="sxs-lookup">
        <span data-stu-id="d6af0-101">The exception that is thrown when the execution stack overflows because it contains too many nested method calls.</span>
      </span>
      <span data-ttu-id="d6af0-102">Diese Klasse kann nicht vererbt werden.</span>
      <span class="sxs-lookup">
        <span data-stu-id="d6af0-102">This class cannot be inherited.</span>
      </span>
    </summary>
    <remarks>
      <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="d6af0-103"><xref:System.StackOverflowException> für Stapelüberlaufs, in der Regel bei der eine sehr tiefe oder unbegrenzte Rekursion ausgelöst.</span><span class="sxs-lookup"><span data-stu-id="d6af0-103"><xref:System.StackOverflowException> is thrown for execution stack overflow errors, typically in case of a very deep or unbounded recursion.</span></span>  
  
 <span data-ttu-id="d6af0-104"><xref:System.StackOverflowException> verwendet das HRESULT COR_E_STACKOVERFLOW mit dem Wert 0x800703E9.</span><span class="sxs-lookup"><span data-stu-id="d6af0-104"><xref:System.StackOverflowException> uses the HRESULT COR_E_STACKOVERFLOW, which has the value 0x800703E9.</span></span> <span data-ttu-id="d6af0-105">Die <xref:System.Reflection.Emit.OpCodes.Localloc> Zwischensprache (IL)-Anweisung löst <xref:System.StackOverflowException>.</span><span class="sxs-lookup"><span data-stu-id="d6af0-105">The <xref:System.Reflection.Emit.OpCodes.Localloc> intermediate language (IL) instruction throws <xref:System.StackOverflowException>.</span></span> <span data-ttu-id="d6af0-106">Eine Liste der anfänglichen Eigenschaftswerte für eine <xref:System.StackOverflowException> Objekt, finden Sie unter den <xref:System.StackOverflowException.%23ctor%2A> Konstruktoren.</span><span class="sxs-lookup"><span data-stu-id="d6af0-106">For a list of initial property values for a <xref:System.StackOverflowException> object, see the <xref:System.StackOverflowException.%23ctor%2A> constructors.</span></span>  
  
## <a name="version-considerations"></a><span data-ttu-id="d6af0-107">Version-Überlegungen</span><span class="sxs-lookup"><span data-stu-id="d6af0-107">Version Considerations</span></span>  
 <span data-ttu-id="d6af0-108">In .NET Framework 1.0 und 1.1 konnten Sie Abfangen einer <xref:System.StackOverflowException> Objekt (z. B. Wiederherstellen einer ungebundenen Rekursion).</span><span class="sxs-lookup"><span data-stu-id="d6af0-108">In the .NET Framework 1.0 and 1.1, you could catch a <xref:System.StackOverflowException> object (for example, to recover from unbounded recursion).</span></span> <span data-ttu-id="d6af0-109">Ab .NET Framework 2.0 können Sie nicht Abfangen einer <xref:System.StackOverflowException> Objekt mit einer `try` / `catch` Block, und den entsprechenden Prozess standardmäßig beendet.</span><span class="sxs-lookup"><span data-stu-id="d6af0-109">Starting with the .NET Framework 2.0, you can’t catch a <xref:System.StackOverflowException> object with a `try`/`catch` block, and the corresponding process is terminated by default.</span></span> <span data-ttu-id="d6af0-110">Daher sollten Sie Ihren Code, um zu erkennen und zu verhindern, dass einen Stapelüberlauf schreiben.</span><span class="sxs-lookup"><span data-stu-id="d6af0-110">Consequently, you should write your code to detect and prevent a stack overflow.</span></span> <span data-ttu-id="d6af0-111">Beispielsweise wenn Ihre app von der Rekursion abhängig ist, verwenden Sie einen Indikator oder ein Zustand die rekursive Schleife zu beenden.</span><span class="sxs-lookup"><span data-stu-id="d6af0-111">For example, if your app depends on recursion, use a counter or a state condition to terminate the recursive loop.</span></span> <span data-ttu-id="d6af0-112">Im folgenden Beispiel wird einen Zähler, um sicherzustellen, dass die Anzahl der rekursiven Aufrufe der `Execute` Methode definiert, durch die Konstante MAX_RECURSIVE_CALLS höchstens nicht überschreiten.</span><span class="sxs-lookup"><span data-stu-id="d6af0-112">The following example uses a counter to ensure that the number of recursive calls to the `Execute` method do not exceed a maximum defined by the MAX_RECURSIVE_CALLS constant.</span></span>  
  
 [!code-csharp[System.StackOverflowException.Class#1](~/samples/snippets/csharp/VS_Snippets_CLR_System/system.stackoverflowexception.class/cs/example1a.cs#1)]
 [!code-vb[System.StackOverflowException.Class#1](~/samples/snippets/visualbasic/VS_Snippets_CLR_System/system.stackoverflowexception.class/vb/example1a.vb#1)]  
  
> [!NOTE]
>  <span data-ttu-id="d6af0-113">Anwenden der <xref:System.Runtime.ExceptionServices.HandleProcessCorruptedStateExceptionsAttribute> -Attribut auf eine Methode, die auslöst, ein <xref:System.StackOverflowException> hat keine Auswirkungen.</span><span class="sxs-lookup"><span data-stu-id="d6af0-113">Applying the <xref:System.Runtime.ExceptionServices.HandleProcessCorruptedStateExceptionsAttribute> attribute to a method that throws a <xref:System.StackOverflowException> has no effect.</span></span> <span data-ttu-id="d6af0-114">Sie können nicht immer noch die Ausnahme im Benutzercode behandeln.</span><span class="sxs-lookup"><span data-stu-id="d6af0-114">You still cannot handle the exception from user code.</span></span>  
  
 <span data-ttu-id="d6af0-115">Wenn Ihre app die common Language Runtime (CLR) hostet, können sie angeben, dass die Anwendungsdomäne, in denen die Stapelüberlauf-Ausnahme tritt auf, und führen Sie den entsprechenden Vorgang weiterhin, die CLR entladen werden soll.</span><span class="sxs-lookup"><span data-stu-id="d6af0-115">If your app hosts the common language runtime (CLR), it can specify that the CLR should unload the application domain where the stack overflow exception occurs and let the corresponding process continue.</span></span> <span data-ttu-id="d6af0-116">Weitere Informationen finden Sie unter [ICLRPolicyManager-Schnittstelle](~/docs/framework/unmanaged-api/hosting/iclrpolicymanager-interface.md).</span><span class="sxs-lookup"><span data-stu-id="d6af0-116">For more information, see [ICLRPolicyManager Interface](~/docs/framework/unmanaged-api/hosting/iclrpolicymanager-interface.md).</span></span>  
  
 ]]></format>
    </remarks>
    <altmember cref="T:System.Exception" />
  </Docs>
  <Members>
    <MemberGroup MemberName=".ctor">
      <AssemblyInfo>
        <AssemblyName>mscorlib</AssemblyName>
        <AssemblyVersion>2.0.5.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <Docs>
        <summary>
          <span data-ttu-id="d6af0-117">Initialisiert eine neue Instanz der <see cref="T:System.StackOverflowException" />-Klasse.</span>
          <span class="sxs-lookup">
            <span data-stu-id="d6af0-117">Initializes a new instance of the <see cref="T:System.StackOverflowException" /> class.</span>
          </span>
        </summary>
      </Docs>
    </MemberGroup>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="public StackOverflowException ();" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig specialname rtspecialname instance void .ctor() cil managed" />
      <MemberSignature Language="DocId" Value="M:System.StackOverflowException.#ctor" />
      <MemberSignature Language="VB.NET" Value="Public Sub New ()" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; StackOverflowException();" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>mscorlib</AssemblyName>
        <AssemblyVersion>1.0.5000.0</AssemblyVersion>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
        <AssemblyVersion>2.0.5.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>netstandard</AssemblyName>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>System.Runtime</AssemblyName>
        <AssemblyVersion>4.2.0.0</AssemblyVersion>
        <AssemblyVersion>4.2.1.0</AssemblyVersion>
      </AssemblyInfo>
      <Parameters />
      <Docs>
        <summary>
          <span data-ttu-id="d6af0-118">Initialisiert eine neue Instanz der <see cref="T:System.StackOverflowException" />-Klasse. Dabei wird die <see cref="P:System.Exception.Message" />-Eigenschaft der neuen Instanz auf eine vom System gelieferte Meldung festgelegt, die den Fehler beschreibt, z. B. "Dieser Vorgang verursachte einen Stapelüberlauf".</span>
          <span class="sxs-lookup">
            <span data-stu-id="d6af0-118">Initializes a new instance of the <see cref="T:System.StackOverflowException" /> class, setting the <see cref="P:System.Exception.Message" /> property of the new instance to a system-supplied message that describes the error, such as "The requested operation caused a stack overflow."</span>
          </span>
          <span data-ttu-id="d6af0-119">Diese Meldung berücksichtigt die aktuelle Systemkultur.</span>
          <span class="sxs-lookup">
            <span data-stu-id="d6af0-119">This message takes into account the current system culture.</span>
          </span>
        </summary>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="d6af0-120">In der folgenden Tabelle werden die anfänglichen Eigenschaftenwerte für eine <xref:System.StackOverflowException>-Instanz aufgeführt.</span><span class="sxs-lookup"><span data-stu-id="d6af0-120">The following table shows the initial property values for an instance of <xref:System.StackOverflowException>.</span></span>  
  
|<span data-ttu-id="d6af0-121">Eigenschaft</span><span class="sxs-lookup"><span data-stu-id="d6af0-121">Property</span></span>|<span data-ttu-id="d6af0-122">Wert</span><span class="sxs-lookup"><span data-stu-id="d6af0-122">Value</span></span>|  
|--------------|-----------|  
|<xref:System.Exception.InnerException%2A>|<span data-ttu-id="d6af0-123">`null`</span><span class="sxs-lookup"><span data-stu-id="d6af0-123">`null`.</span></span>|  
|<xref:System.Exception.Message%2A>|<span data-ttu-id="d6af0-124">Die lokalisierte Zeichenfolge mit der Fehlermeldung.</span><span class="sxs-lookup"><span data-stu-id="d6af0-124">The localized error message string.</span></span>|  
  
 ]]></format>
        </remarks>
      </Docs>
    </Member>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="public StackOverflowException (string message);" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig specialname rtspecialname instance void .ctor(string message) cil managed" />
      <MemberSignature Language="DocId" Value="M:System.StackOverflowException.#ctor(System.String)" />
      <MemberSignature Language="VB.NET" Value="Public Sub New (message As String)" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; StackOverflowException(System::String ^ message);" />
      <MemberSignature Language="F#" Value="new StackOverflowException : string -&gt; StackOverflowException" Usage="new System.StackOverflowException message" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>mscorlib</AssemblyName>
        <AssemblyVersion>1.0.5000.0</AssemblyVersion>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
        <AssemblyVersion>2.0.5.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>netstandard</AssemblyName>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>System.Runtime</AssemblyName>
        <AssemblyVersion>4.2.0.0</AssemblyVersion>
        <AssemblyVersion>4.2.1.0</AssemblyVersion>
      </AssemblyInfo>
      <Parameters>
        <Parameter Name="message" Type="System.String" />
      </Parameters>
      <Docs>
        <param name="message">
          <span data-ttu-id="d6af0-125">Ein <see cref="T:System.String" />, der den Fehler beschreibt.</span>
          <span class="sxs-lookup">
            <span data-stu-id="d6af0-125">A <see cref="T:System.String" /> that describes the error.</span>
          </span>
          <span data-ttu-id="d6af0-126">Der Inhalt der Meldung soll in verständlicher Sprache gehalten sein.</span>
          <span class="sxs-lookup">
            <span data-stu-id="d6af0-126">The content of message is intended to be understood by humans.</span>
          </span>
          <span data-ttu-id="d6af0-127">Der Aufrufer dieses Konstruktors muss sicherstellen, dass diese Zeichenfolge für die aktuelle Systemkultur lokalisiert wurde.</span>
          <span class="sxs-lookup">
            <span data-stu-id="d6af0-127">The caller of this constructor is required to ensure that this string has been localized for the current system culture.</span>
          </span>
        </param>
        <summary>
          <span data-ttu-id="d6af0-128">Initialisiert eine neue Instanz der <see cref="T:System.StackOverflowException" />-Klasse mit einer angegebenen Fehlermeldung.</span>
          <span class="sxs-lookup">
            <span data-stu-id="d6af0-128">Initializes a new instance of the <see cref="T:System.StackOverflowException" /> class with a specified error message.</span>
          </span>
        </summary>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="d6af0-129">In der folgenden Tabelle werden die anfänglichen Eigenschaftenwerte für eine <xref:System.StackOverflowException>-Instanz aufgeführt.</span><span class="sxs-lookup"><span data-stu-id="d6af0-129">The following table shows the initial property values for an instance of <xref:System.StackOverflowException>.</span></span>  
  
|<span data-ttu-id="d6af0-130">Eigenschaft</span><span class="sxs-lookup"><span data-stu-id="d6af0-130">Property</span></span>|<span data-ttu-id="d6af0-131">Wert</span><span class="sxs-lookup"><span data-stu-id="d6af0-131">Value</span></span>|  
|--------------|-----------|  
|<xref:System.Exception.InnerException%2A>|<span data-ttu-id="d6af0-132">Ein NULL-Verweis (`Nothing` in Visual Basic).</span><span class="sxs-lookup"><span data-stu-id="d6af0-132">A null reference (`Nothing` in Visual Basic).</span></span>|  
|<xref:System.Exception.Message%2A>|<span data-ttu-id="d6af0-133">Die Zeichenfolge der Fehlermeldung.</span><span class="sxs-lookup"><span data-stu-id="d6af0-133">The error message string.</span></span>|  
  
 ]]></format>
        </remarks>
      </Docs>
    </Member>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="public StackOverflowException (string message, Exception innerException);" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig specialname rtspecialname instance void .ctor(string message, class System.Exception innerException) cil managed" />
      <MemberSignature Language="DocId" Value="M:System.StackOverflowException.#ctor(System.String,System.Exception)" />
      <MemberSignature Language="VB.NET" Value="Public Sub New (message As String, innerException As Exception)" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; StackOverflowException(System::String ^ message, Exception ^ innerException);" />
      <MemberSignature Language="F#" Value="new StackOverflowException : string * Exception -&gt; StackOverflowException" Usage="new System.StackOverflowException (message, innerException)" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>mscorlib</AssemblyName>
        <AssemblyVersion>1.0.5000.0</AssemblyVersion>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
        <AssemblyVersion>2.0.5.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>netstandard</AssemblyName>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>System.Runtime</AssemblyName>
        <AssemblyVersion>4.2.0.0</AssemblyVersion>
        <AssemblyVersion>4.2.1.0</AssemblyVersion>
      </AssemblyInfo>
      <Parameters>
        <Parameter Name="message" Type="System.String" />
        <Parameter Name="innerException" Type="System.Exception" />
      </Parameters>
      <Docs>
        <param name="message">
          <span data-ttu-id="d6af0-134">Die Fehlermeldung, in der die Ursache der Ausnahme erklärt wird.</span>
          <span class="sxs-lookup">
            <span data-stu-id="d6af0-134">The error message that explains the reason for the exception.</span>
          </span>
        </param>
        <param name="innerException">
          <span data-ttu-id="d6af0-135">Die Ausnahme, die die Ursache der aktuellen Ausnahme ist.</span>
          <span class="sxs-lookup">
            <span data-stu-id="d6af0-135">The exception that is the cause of the current exception.</span>
          </span>
          <span data-ttu-id="d6af0-136">Wenn der Parameter <c>innerException</c> kein NULL-Verweis ist (<see langword="Nothing" /> in Visual Basic), wird die aktuelle Ausnahme in einem <see langword="catch" />-Block ausgelöst, der die innere Ausnahme behandelt.</span>
          <span class="sxs-lookup">
            <span data-stu-id="d6af0-136">If the <c>innerException</c> parameter is not a null reference (<see langword="Nothing" /> in Visual Basic), the current exception is raised in a <see langword="catch" /> block that handles the inner exception.</span>
          </span>
        </param>
        <summary>
          <span data-ttu-id="d6af0-137">Initialisiert eine neue Instanz der <see cref="T:System.StackOverflowException" />-Klasse mit einer angegebenen Fehlermeldung und einem Verweis auf die innere Ausnahme, die diese Ausnahme ausgelöst hat.</span>
          <span class="sxs-lookup">
            <span data-stu-id="d6af0-137">Initializes a new instance of the <see cref="T:System.StackOverflowException" /> class with a specified error message and a reference to the inner exception that is the cause of this exception.</span>
          </span>
        </summary>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="d6af0-138">Eine Ausnahme, die als direktes Ergebnis einer vorhergehenden Ausnahme ausgelöst wird, kann in der <xref:System.Exception.InnerException%2A>-Eigenschaft über einen Verweis auf die vorhergehende Ausnahme verfügen.</span><span class="sxs-lookup"><span data-stu-id="d6af0-138">An exception that is thrown as a direct result of a previous exception can include a reference to the previous exception in the <xref:System.Exception.InnerException%2A> property.</span></span> <span data-ttu-id="d6af0-139">Die <xref:System.Exception.InnerException%2A>-Eigenschaft gibt denselben Wert zurück, der an den Konstruktor übergeben wurde, bzw. einen NULL-Verweis (`Nothing` in Visual Basic), wenn die <xref:System.Exception.InnerException%2A>-Eigenschaft nicht den Wert für die innere Ausnahme für den Konstruktor bereitstellt.</span><span class="sxs-lookup"><span data-stu-id="d6af0-139">The <xref:System.Exception.InnerException%2A> property returns the same value that is passed into the constructor, or a null reference (`Nothing` in Visual Basic) if the <xref:System.Exception.InnerException%2A> property does not supply the inner exception value to the constructor.</span></span>  
  
 <span data-ttu-id="d6af0-140">In der folgenden Tabelle werden die anfänglichen Eigenschaftenwerte für eine <xref:System.StackOverflowException>-Instanz aufgeführt.</span><span class="sxs-lookup"><span data-stu-id="d6af0-140">The following table shows the initial property values for an instance of <xref:System.StackOverflowException>.</span></span>  
  
|<span data-ttu-id="d6af0-141">Eigenschaft</span><span class="sxs-lookup"><span data-stu-id="d6af0-141">Property</span></span>|<span data-ttu-id="d6af0-142">Wert</span><span class="sxs-lookup"><span data-stu-id="d6af0-142">Value</span></span>|  
|--------------|-----------|  
|<xref:System.Exception.InnerException%2A>|<span data-ttu-id="d6af0-143">Der Verweis auf die interne Ausnahme.</span><span class="sxs-lookup"><span data-stu-id="d6af0-143">The inner exception reference.</span></span>|  
|<xref:System.Exception.Message%2A>|<span data-ttu-id="d6af0-144">Die Zeichenfolge der Fehlermeldung.</span><span class="sxs-lookup"><span data-stu-id="d6af0-144">The error message string.</span></span>|  
  
 ]]></format>
        </remarks>
        <altmember cref="T:System.Exception" />
      </Docs>
    </Member>
  </Members>
</Type>